### PR TITLE
ROCm 5.0.2 compatibility

### LINKED ISSUE #36: Build fails 2.8.0-1
The build stops with the following error. I use the newest 5.0.2-1 rocm packages. Any ideas how to resolve this problem?

```
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/XNNPACK/archive/113092317754c7dea47bfb3cb49c4f59c3c1fa10.zip failed: class java.io.FileNotFoundException GET returned 404 Not Found
INFO: Analyzed 4 targets (280 packages loaded, 22789 targets configured).
INFO: Found 4 targets...
ERROR: /tmp/makepkg/tensorflow-rocm/src/tensorflow-2.8.0-rocm/tensorflow/core/kernels/BUILD:4987:18: Compiling tensorflow/core/kernels/sparse_tensor_dense_matmul_op_gpu.cu.cc failed: undeclared inclusion(s) in rule '//tensorflow/core/kernels:sparse_tensor_dense_matmul_op_gpu':
this rule is missing dependency declarations for the following files included by 'tensorflow/core/kernels/sparse_tensor_dense_matmul_op_gpu.cu.cc':
  '/opt/rocm/llvm/lib/clang/14.0.0/include/__clang_hip_runtime_wrapper.h'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/stddef.h'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/__clang_hip_libdevice_declares.h'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/__clang_hip_math.h'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/cuda_wrappers/algorithm'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/cuda_wrappers/new'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/limits.h'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/stdint.h'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/__clang_cuda_math_forward_declares.h'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/__clang_hip_cmath.h'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/__clang_cuda_complex_builtins.h'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/cuda_wrappers/complex'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/__stddef_max_align_t.h'
  '/opt/rocm/llvm/lib/clang/14.0.0/include/stdarg.h'
clang-14: warning: argument unused during compilation: '-fcuda-flush-denormals-to-zero' [-Wunused-command-line-argument]
INFO: Elapsed time: 859.146s, Critical Path: 55.36s
INFO: 4084 processes: 413 internal, 3671 local.
FAILED: Build did NOT complete successfully
==> ERROR: A failure occurred in build().
```

I was also missing the file /opt/rocm/.info/version-dev so i linked it manually. Also some files were missing so i needed to install the following packages from https://github.com/rocm-arch/rocm-arch. Only with these changes the build process would start.

```
ln -s  /opt/rocm/.info/version /opt/rocm/.info/version-dev

packages:
rocand
hipfft
rocsparse
hipsparse
rccl
```

### PR DESCRIPTION
**Pull request by RavenOchlich**
Tensorflow version 2.8.0 doesn't seem to be compatible with ROCm 5.0.2. So this PKGBUILD uses the ROCmSoftwarePlatform / tensorflow-upstream git as source.  The package depends on miopen 5.0.2 which is currently not up to date, but a pull request for 5.0.2 is pending.

Closes #36

### CODE REVIEW

File: PKGBUILD [hunk 1]
```diff
@@ -12,7 +12,6 @@ _build_opt=1
 
 pkgname=()
 [ "$_build_no_opt" -eq 1 ] && pkgname+=(tensorflow-rocm python-tensorflow-rocm)
-[ "$_build_opt" -eq 1 ] && pkgname+=(tensorflow-opt-rocm python-tensorflow-opt-rocm)
```
Comment (acxz) [inline PKGBUILD:15]: Why is this line deleted? Can you add it back?

File: PKGBUILD [hunk 2]
```diff
@@ -25,13 +24,16 @@ depends=('c-ares' 'intel-mkl' 'onednn' 'pybind11' 'openssl' 'lmdb' 'libpng' 'cur
 makedepends=('bazel' 'python-numpy' 'rocm-hip-sdk' 'miopen' 'rccl' 'git'
              'python-pip' 'python-wheel' 'python-setuptools' 'python-h5py'
              'python-keras-applications' 'python-keras-preprocessing'
-             'cython')
+             'cython' 'rocblas' 'rocrand' 'rocfft' 'hipfft' 'roctracer'
+             'hipsparse' 'hipsolver' 'rocsolver' 'rocm-hip-sdk')
```
Comment (acxz) [inline PKGBUILD]: Are these packages not included in rocm-hip-sdk which is already listed?

File: PKGBUILD [hunk 3]
```diff
@@ -25,13 +24,16 @@ depends=('c-ares' 'intel-mkl' 'onednn' 'pybind11' 'openssl' 'lmdb' 'libpng' 'cur
...
 optdepends=('tensorboard: Tensorflow visualization toolkit')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/tensorflow/tensorflow/archive/v${_pkgver}.tar.gz"
-        fix-c++17-compat.patch)
+source=("$pkgname::git+https://github.com/ROCmSoftwarePlatform/tensorflow-upstream.git"
```
Comment (acxz) [inline PKGBUILD:29]: This is the release version of the package therefore this URL should not be changed.

File: PKGBUILD [hunk 4]
```diff
@@ -25,13 +24,16 @@ depends=('c-ares' 'intel-mkl' 'onednn' 'pybind11' 'openssl' 'lmdb' 'libpng' 'cur
...
+        fix-c++17-compat.patch
+        hiprand.patch)
```
Comment (acxz) [inline PKGBUILD:31]: Can you make this patch pull from a pull request instead of manually having the patch in this repo? Having the patch sent upstream helps us verify it via the upstream devs and ensure it gets merged in eventually, where the patch is not even necessary.

**Review by acxz — CHANGES_REQUESTED** [review]
When updating the PKGBUILD make sure to bump the pkgrel and also to update the .SRCINFO file.

File: PKGBUILD (hunk 3 above)
Comment (RavenOchlich) [reply PKGBUILD:29]: Then we need to wait or i need to create a git version because there is no proper release yet. https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/tags

File: PKGBUILD (hunk 2 above)
Comment (RavenOchlich) [reply PKGBUILD]: The remaining packages in my new commit are not included in rocm-hip-sdk.

File: PKGBUILD (hunk 1 above)
Comment (RavenOchlich) [reply PKGBUILD:15]: I took it out because the compile took very long and i didn't figure out how to inhibit the opt version from building. Am i missing something?

File: PKGBUILD (hunk 3 above)
Comment (acxz) [reply PKGBUILD:29]: Nice!, Should rocm-hip-sdk be removed as well since it is listed earlier also can you add roctracer and hipsolver next to the other rocm deps after rccl on the same line

File: PKGBUILD (hunk 1 above)
Comment (acxz) [reply PKGBUILD:15]: Yes the PKGBUILD should build both packages, when testing you can build just one. For an easy bypass just change the variable above to 0 that you do not want. i.e. _build_opt=0 on line 11.

File: PKGBUILD (hunk 2 above)
Comment (acxz) [reply PKGBUILD]: Nice!, Should rocm-hip-sdk be removed as well since it is listed earlier also can you add roctracer and hipsolver next to the other rocm deps after rccl on the same line

File: PKGBUILD (hunk 3 above)
Comment (acxz) [reply PKGBUILD:29]: This PKGBUILD pulls from upstream tensorflow not from AMD's tensorflow fork. We should wait or add patches like u have done for hiprand.

Comment (acxz) [reply PKGBUILD:29]: Even a git version of this PKGBUILD will use upstream tensorflow and not AMD's fork.
What you are suggesting is something like `tensorflow-rocm-amd-git`

File: PKGBUILD (hunk 4 above)
Comment (RavenOchlich) [reply PKGBUILD:31]: I need to investigate that further, because i am not shure if the arch packaging is wrong, or if it's a tensorflow problem.

File: PKGBUILD (hunk 3 above)
Comment (acxz) [reply PKGBUILD:29]: thx!

**Comment by acxz** [comment]
Closing due to inactivity
